modules/gapi/test/common/gapi_video_tests_inl.hpp

```diff
@@ -131,7 +131,51 @@ TEST_P(BackgroundSubtractorTest, AccuracyTest)
     // Allowing 1% difference of all pixels between G-API and reference OpenCV results
     testBackgroundSubtractorStreaming(gapiBackSub, pOCVBackSub, 1, 1, learningRate, testNumFrames);
 }
+
+TEST_P(KalmanFilterTest, AccuracyTest)
+{
+    cv::gapi::video::KalmanParams kp(dDim, mDim, cDim, depth);
+
+    //measurement vector
+    cv::Mat measure_vec(mDim, 1, depth);
+    cv::randu(measure_vec, Scalar::all(-1), Scalar::all(1));
+
+    //control vector
+    cv::Mat ctrl_vec = Mat::zeros(cDim > 0 ? cDim : dDim, 1, depth);
```

Comment:
It is irrelevant and very inconvenient for user to create and initialize the `control` vector even in the case `cDim = 0` which is the case when user don't even intend to use the controlling force at all.

Here is what I propose.
We should restrict `cDim = 0` in our kernel. Actually, in its main overload.
Then, we should add an overload without the 3rd argument `control` and demand `cDim = 0` there which would mean the system isn't under the controlling force.

This will clarify both our kernel implementation and its usage.
Please let's discuss (mb with the team even)

Reply:
Overload was added